internal/command/jsonplan/values.go

```diff
@@ -98,18 +98,21 @@ func marshalPlannedOutputs(changes *plans.Changes) (map[string]Output, error) {
 func marshalPlannedValues(changes *plans.Changes, schemas *tofu.Schemas) (Module, error) {
 	var ret Module
 
-	// build two maps:
+	// build three maps:
 	// 		module name -> [resource addresses]
 	// 		module -> [children modules]
+	// 		resource addr -> change src
 	moduleResourceMap := make(map[string][]addrs.AbsResourceInstance)
 	moduleMap := make(map[string][]addrs.ModuleInstance)
+	changeMap := make(map[string]*plans.ResourceInstanceChangeSrc)
 	seenModules := make(map[string]bool)
 
 	for _, resource := range changes.Resources {
 		// If the resource is being deleted, skip over it.
 		// Deposed instances are always conceptually a destroy, but if they
 		// were gone during refresh then the change becomes a noop.
 		if resource.Action != plans.Delete && resource.DeposedKey == states.NotDeposed {
+			changeMap[resource.Addr.String()] = resource
```

Comment:
nit: I would prefer `changeMap` content to be equal to what `plans.Changes.ResourceInstance` returns, i.e. all `NotDeposed` changes. Then, it makes sense to rename `changeMap` to what reflects its content better, e.g. `notDeposedChangeMap`. This way we keep the original behaviour and make the retrieval more efficient. WDYT?

Reply:
Makes sense, I agree. Flipped some conditionals to avoid excessive indentation as well, thanks.